### DISCUSSION TITLE
Is it a way to enable field assign to not scalar types?

### DISCUSSION THREAD

**Discussion by CursedMun** [MichalLytek/typegraphql-prisma#363] [Q&A] [answered]
I have a generated model that creates @Typegraphql.field for scalar types but doesn't do the same for class types
```
@TypeGraphQL.Field(_type => String, {
    nullable: true
  })
  slug?: string | null;
  //here we don't have typegraphql.field(_type => ArticleAuthors, {nullable:true})
  authors?: ArticleAuthors[]; 
```

**Comment by MichalLytek** [comment]
Relation fields require some logic to fetch them. Declaring it in object-type won't automagically populate those fields when you execute the query.

That's why those fields are registered by field resolvers of generated relation resolvers:
https://prisma.typegraphql.com/docs/basics/usage#relations-resolvers

**Reply by CursedMun** [reply]
I understand that they won't fetch 'automagically' as you say, I need them as a generated output model, I have the logic behind the scene. And for me, it doesn't make any sense making relation resolvers

**Comment by CursedMun** [comment, marked as answer]
I found a workaround for this situation for someone coming here with the same problem just remove the following line inside node-modules/typegraphql-prisma/model-type-class.js  line 67 
```
field.relationName ||
```
